lib/node/fullnode.js

```diff
@@ -194,6 +194,16 @@ class FullNode extends Node {
   init() {
     // Bind to errors
     this.chain.on('error', err => this.error(err));
+    this.chain.on('abort', (msg) => {
```

Comment:
This needs to be `async` otherwise `this.close` would not work.
We make sure we don't get `unhandledRejection` by using `try/catch`.

also `await` before `this.close`.

Reply:
I'm confused https://github.com/handshake-org/hsd/pull/650#discussion_r863638293

Can we add await to the listener if its not asycemitter?

Reply:
yeah, we can. It's just if that promise failed (the async method itself) we would get unhandledRejection because there's no one there to catch those errors, that's bad behaviour. But if we make sure promise is handled nicely. E.g. if I rewrote this in this form:

```javascript
this.chain.on('abort', (msg) => {
  this.close.catch((e) => {
    // ...
    // process.exit(INTERNAL_ERROR);
  });
});
```

I think it becomes apparent why we can do this and why we absolutely need try/catch in there. `async/try/catch` is just nicer way to write same as above.